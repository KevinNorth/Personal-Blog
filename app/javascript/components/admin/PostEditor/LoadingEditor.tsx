import React from 'react';
import { Col, Container, Form, Placeholder, Row } from 'react-bootstrap';

function LoadingEditor(): React.ReactElement {
  return (
    <Container fluid>
      <Form>
        <Row>
          <Col xs={12}>
            <Placeholder size="lg" animation="glow" className="w-100" />
          </Col>
        </Row>
        <Row>
          <Col xs={6}>
            <Placeholder animation="glow" className="w-100" />
          </Col>
          <Col xs={6}>
            <Placeholder animation="glow" className="w-100" />
          </Col>
        </Row>
        <Row>
          <Col xs={12}>
            <Placeholder animation="glow" className="w-100" />
          </Col>
        </Row>
        <Row>
          <Col xs={4}>
            <Placeholder animation="glow" className="w-100" />
          </Col>
          <Col xs={8}>
            <Placeholder animation="glow" className="w-100" />
          </Col>
        </Row>
        <Row>
          <Col xs={12}>
            <Placeholder animation="glow" className="w-100" />
          </Col>
        </Row>
      </Form>
    </Container>
  );
}

export default LoadingEditor;
