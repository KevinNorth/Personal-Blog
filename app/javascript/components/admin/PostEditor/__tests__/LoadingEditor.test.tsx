/**
 * @jest-environment jsdom
 */
import React from 'react';
import snapshotRenderer from 'react-test-renderer';
import { act } from '@testing-library/react';
import LoadingEditor from '../LoadingEditor';

describe('LoadingEditor', () => {
  it('renders as expected', () => {
    let wrapper;
    act(() => {
      wrapper = snapshotRenderer.create(<LoadingEditor />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
