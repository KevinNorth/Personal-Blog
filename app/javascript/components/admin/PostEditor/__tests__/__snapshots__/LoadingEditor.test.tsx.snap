// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingEditor renders as expected 1`] = `
<div
  className="container-fluid"
>
  <form
    className=""
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <span
          className="w-100 placeholder-glow placeholder-lg"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <span
          className="w-100 placeholder-glow"
        />
      </div>
      <div
        className="col-6"
      >
        <span
          className="w-100 placeholder-glow"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <span
          className="w-100 placeholder-glow"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-4"
      >
        <span
          className="w-100 placeholder-glow"
        />
      </div>
      <div
        className="col-8"
      >
        <span
          className="w-100 placeholder-glow"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <span
          className="w-100 placeholder-glow"
        />
      </div>
    </div>
  </form>
</div>
`;
